code/ACLoader.cpp

```diff
@@ -584,9 +587,19 @@ aiNode* AC3DImporter::ConvertObjectSection(Object& object,
 
                 // allocate storage for vertices and normals
                 mesh->mNumFaces = (*cit).first;
+                if (mesh->mNumFaces == 0) {
+                    throw DeadlyImportError("AC3D: No faces");
+                } else if (mesh->mNumFaces > std::numeric_limits<int32_t>::max() / sizeof(aiFace)) {
```

Comment:
Actually it's 32 bits since it uses signed int. I more or less picked it out of a hat.

The reason for both the check and and strange limit is that Valgrind and AddressSanitizer don't really handle running out of memory (or address space) and just crash the program. This makes debugging harder and gives false positives when fuzzing.

This is used in several places. By putting in one place it becomes easy to change (to whatever we think is good). Where should we put the general version and what should it be called? I was thinking something like
\#define AI_MAX_MEMORY_LIMIT(type) (bignumber / sizeof(type))